GameEngine/GameEngine/Engine/Graphic/DRBLightElementManager.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <memory>
+
+#include "Utils/ObjectPool.hpp"
+#include "Utils/Dependency.hpp"
+
+#include "DRBPointLight.hpp"
+#include "DRBSpotLight.hpp"
+
+namespace AGE
+{
+	class BFCCullableHandle;
+	class BFCBlockManagerFactory;
+	class IProperty;
+
+	// That's a SCENE dependency, and not a global one, set on per scene
+	class DRBLightElementManager : public Dependency<DRBLightElementManager>
+	{
+	public:
+		DRBLightElementManager(BFCBlockManagerFactory *factory);
+		BFCCullableHandle addPointLight();
+		// Be carefull, there is to mush properties to pass in parametter, so you have to add Properties, manually
+		// after the handle creation
+		BFCCullableHandle addSpotLight();
+		void removePointLight(BFCCullableHandle &handle);
+		void removeSpotLight(BFCCullableHandle &handle);
+	private:
+		BFCBlockManagerFactory *_bfcBlockManager = nullptr;
+		ObjectPool<DRBPointLight> _pointLightPool;
+		ObjectPool<DRBSpotLight> _spotLightPool;
+	};
+}
```

Comment:
This is supposed contain the light for rendering and culled it with multi-threading ?